contracts/CreditorProxy.sol

```diff
@@ -0,0 +1,350 @@
+            totalCreditorPayment
+        )) {
+            LogError(
+                uint8(Errors.CREDITOR_BALANCE_OR_ALLOWANCE_INSUFFICIENT),
+                creditor,
+                creditorCommitmentHash
+            );
+            return NULL_ISSUANCE_HASH;
+        }
+
+        // Transfer principal from creditor to creditorProxy
+        if (totalCreditorPayment > 0) {
+            require(transferTokensFrom(
+                orderAddresses[4],
+                creditor,
+                address(this),
+                totalCreditorPayment
+            ));
+        }
+
+        // Fill debt order with this contract playing the role of creditor
+        bytes32 agreementId = contractRegistry.debtKernel().fillDebtOrder(
+            address(this),
+            orderAddresses,
+            orderValues,
+            orderBytes32,
+            signaturesV,
+            signaturesR,
+            signaturesS
+        );
+
+        require(agreementId != NULL_ISSUANCE_HASH);
+
+        debtOfferFilled[creditorCommitmentHash] = true;
+
+        // transfer debt token to real creditor
+        contractRegistry.debtToken().transfer(creditor, uint256(agreementId));
+
+        LogDebtOfferFilled(creditor, creditorCommitmentHash, agreementId);
+
+        return agreementId;
+    }
+
+    /**
+     * Allows creditor to prevent a debt offer from being used in the future
+     */
+    function cancelDebtOffer(
+        address[4] commitmentAddresses,
+        uint[4] commitmentValues,
+        bytes32[1] termsContractParameters
+    )
+        public
+        whenNotPaused
+    {
+        // sender must be the creditor
+        require(msg.sender == commitmentAddresses[0]);
+        bytes32 creditorCommitmentHash = getCreditorCommitmentHash(
+            commitmentAddresses,
+            commitmentValues,
+            termsContractParameters
+        );
+        debtOfferCancelled[creditorCommitmentHash] = true;
+        LogDebtOfferCancelled(commitmentAddresses[0], creditorCommitmentHash);
+    }
+
+    ////////////////////////
+    // INTERNAL FUNCTIONS //
+    ////////////////////////
+
+    /**
+     * Returns the messaged signed by the creditor to indicate their commitment
+     */
+    function getCreditorCommitmentHash(
+        address[4] commitmentAddresses,
+        uint[4] commitmentValues,
+        bytes32[1] termsContractParameters
+    )
+        internal
+        pure
+        returns (bytes32 _creditorCommitmentHash)
+    {
+        return keccak256(
+            commitmentAddresses[0], // creditor
+            commitmentAddresses[1], // repayment router version
+            commitmentValues[0], // creditor fee
+            commitmentAddresses[2], // underwriter
+            commitmentValues[1], // underwriterRiskRating
+            commitmentAddresses[3], // termsContract
+            termsContractParameters[0],
+            commitmentValues[2], // commitmentExpirationTimestampInSec
+            commitmentValues[3] // salt
```

Comment:
Question here for curious ☺. Why do we not need principalToken and principalAmount field? Do we mean that creditor need not sign this two field in his commitment?